packages/internal/src/generate/watch.ts

```diff
@@ -28,67 +29,190 @@ import {
 
-    const absPath = path.join(rwjsPaths.base, p)
+    eventName = eventName as 'add' | 'change' | 'unlink' | 'unlinkDir'
 
-    if (absPath.indexOf('Cell') !== -1 && isCellFile(absPath)) {
-      await generateTypeDefGraphQLWeb()
-      if (eventName === 'unlink') {
-        fs.unlinkSync(mirrorPathForCell(absPath, rwjsPaths)[0])
-      } else {
-        generateMirrorCell(absPath, rwjsPaths)
+    const absPath = path.join(rwjsPaths.web.base, p)
+
+    if (eventName === 'unlinkDir') {
+      if (absPath.endsWith('Cell')) {
+        const mirrorDir = path.join(
+          rwjsPaths.generated.types.mirror,
+          path.relative(rwjsPaths.base, p)
+        )
+
+        try {
+          fse.remove(mirrorDir)
+        } catch {
+          // Ignore
+        }
       }
 
+      return
+    }
+
+    if (eventName === 'unlink') {
+      // When unlinking (i.e. removing or renaming) a file we can't use
+      // `isCellFile()` because that function needs to look inside the file to
+      // determine if it's a cell or not, and it can't do that on a file that
+      // doesn't exist anymore
+
+      const { name } = path.parse(p)
+
+      if (name.endsWith('Cell')) {
+        // Probably a cell, but we can never be sure. But if it was indeed a
+        // cell it should have had a mirror cell that we now need to remove
+        try {
+          fs.unlinkSync(mirrorPathForCell(absPath, rwjsPaths)[0])
+
+          console.log(action[eventName], 'Cell:', '\x1b[2m', p, '\x1b[0m')
+        } catch {
+          // Unlinking will fail if the mirror cell doesn't exist. Just ignore
+          // the exception in that case.
+          // Unlinking can also fail, especially on Windows, if the file is
+          // open somewhere, like in some editor window. Let's just ignore that
+          // as well for now
+        }
+      }
+    }
+
+    if (eventName !== 'unlink' && isCellFile(absPath)) {
+      await generateTypeDefGraphQLWeb()
+      generateMirrorCell(absPath, rwjsPaths)
+
       console.log(action[eventName], 'Cell:', '\x1b[2m', p, '\x1b[0m')
     } else if (absPath === rwjsPaths.web.routes) {
       generateTypeDefRouterRoutes()
       console.log(action[eventName], 'Routes:', '\x1b[2m', p, '\x1b[0m')
-    } else if (absPath.indexOf('Page') !== -1 && isPageFile(absPath)) {
+    } else if (isPageFile(absPath)) {
       generateTypeDefRouterPages()
       console.log(action[eventName], 'Page:', '\x1b[2m', p, '\x1b[0m')
     } else if (isDirectoryNamedModuleFile(absPath)) {
+      // Remember that files ending with Cell can also be just a regular
+      // Directory Named Module, for example <TableCell> used as wrapper for
+      // simple <td>s. If <TableCell> lived in /TableCell/TableCell.tsx it'd
+      // be a Directory Named Module and it's mirror needs to be deleted
+
       if (eventName === 'unlink') {
-        fs.unlinkSync(mirrorPathForDirectoryNamedModules(absPath, rwjsPaths)[0])
+        try {
+          fs.unlinkSync(
```

Comment:
Should we switch to fse here too? 

There was a reason we used fse before - I can't remember now though. Maybe there's a comment I left somewhere explaining why

Reply:
I could probably switch to fse. But I don't have to. This is only a single file (`index.d.ts`) so the regular `fs.unlinkSync` function works well. I only had to use fse when I wanted to remove a full directory with all the files inside.

But please let me know if you want me to switch to fse. Happy to make the change.

Reply:
All good - I noticed we used fse elsewhere, so might as well use it here, but up to you, no big deal.